lib/AST/SubstitutionMap.cpp

```diff
@@ -355,8 +355,15 @@ SubstitutionMap::lookupConformance(CanType type, ProtocolDecl *proto) const {
   // Check whether the superclass conforms.
   if (auto superclass = genericSig->getSuperclassBound(type)) {
     LookUpConformanceInSignature lookup(getGenericSignature().getPointer());
-    if (auto conformance = lookup(type->getCanonicalType(), superclass, proto))
+    if (auto conformance = lookup(type->getCanonicalType(), superclass, proto)){
```

Comment:
LookUpConformanceInSignature just does a ModuleDecl::lookupConformance(). Why doesn't that do the right thing? Should we instead just `s/type/substType/` here?

Reply:
It seems to do more than that. `type` here appears to intentionally be referring to the unsubstituted type.

Reply:
Ah, right.

So I think instead you want `s/superclass/substType/` in the `lookup()` call.

`LookUpConformanceInSignature::operator()` ignores `dependentType` (the original type) and calls `GenericSignature::lookupConformance()` with `conformingReplacementType` (the substituted type).

The issue is that we're calling `operator()` (via `lookup()`) with `superclass` as the `conformingReplacementType`. I think `superclass` should be `substType` here.

Reply:
OK, I'll try that.